dashboards-observability/public/components/visualizations/charts/stats/stats.tsx

```diff
@@ -0,0 +1,448 @@
+            ...STATS_ANNOTATION,
+            xanchor: 'left',
+            yanchor: 'bottom',
+            text: label,
+            font: {
+              size: titleSize,
+              color: isDarkMode ? COLOR_WHITE : COLOR_BLACK,
+              family: 'Roboto',
+            },
+            x: index / metricsLength + ANNOTATION_MARGIN_LEFT,
+            y: 1,
+            metricValue: value,
+            type: 'name',
+          },
+          {
+            ...STATS_ANNOTATION,
+            xanchor: 'left',
+            yanchor: 'top',
+            text: createValueText(value),
+            font: {
+              size: valueSize,
+              color: valueColor,
+              family: 'Roboto',
+            },
+            x: index / metricsLength + ANNOTATION_MARGIN_LEFT,
+            y: 1,
+            type: 'value',
+            metricValue: value,
+          },
+        ]
+      : [
+          {
+            ...STATS_ANNOTATION,
+            x:
+              metricsLength === 1
+                ? 0.5
+                : index === 0
+                ? ((1 / metricsLength) * 1) / 2
+                : (index + 1) / metricsLength - ((1 / metricsLength) * 1) / 2,
+            xanchor: 'center',
+            y: 0.95,
+            yanchor: 'bottom',
+            text: textMode === 'values' ? createValueText(value) : label,
+            font: {
+              size: textMode === 'values' ? valueSize : titleSize,
+              color: textMode === 'names' ? (isDarkMode ? COLOR_WHITE : COLOR_BLACK) : valueColor,
+              family: 'Roboto',
+            },
+            type: textMode === 'names' ? 'name' : 'value',
+            metricValue: value,
+          },
+        ];
+
+  // extend y axis range to increase height of subplot w.r.t metric data
+  const extendYaxisRange = (metric: ConfigListEntry) => {
+    const sortedData = data[metric.label].slice().sort((curr: number, next: number) => next - curr);
+    const avgSeriesDiff = sortedData.slice(0, 5).reduce(function (prev, curr, index: number) {
+      if (data[metric.label][index + 1])
+        return (prev += Math.abs(Number((data[metric.label][index + 1] - curr).toFixed(2))));
+      return prev;
+    }, 0);
+    return sortedData[0] + (avgSeriesDiff ? avgSeriesDiff : 100);
+  };
+
+  const generateLineTraces = () => {
+    return metrics.map((metric: ConfigListEntry, metricIndex: number) => {
+      autoChartLayout = {
+        ...autoChartLayout,
+        annotations: autoChartLayout.annotations.concat(
+          orientation === DefaultOrientation || metricsLength === 1
+            ? createAnnotationVerticalOrientation({
+                label: metric.label,
+                value:
+                  typeof data[metric.label][data[metric.label].length - 1] === 'number'
+                    ? getRoundOf(data[metric.label][data[metric.label].length - 1], precisionValue)
+                    : 0,
+                index: metricIndex,
+                valueColor: '',
+              })
```

Comment:
This complete object can be defined outside, for example: as the first statement inside this map callback and can be used something like:
```JS
DefaultOrientation || metricsLength === 1
  ? createAnnotationVerticalOrientation(annotationOption)
  :  createAnnotationsHorizontalOrientation(annotationOption)
```
This is very easy to read as compared to the complete object inside the ternary operator.